backend/internal/service/handler/users/user.go

```diff
@@ -110,13 +122,22 @@ func (h *Handler) GetUserProfile(c *fiber.Ctx) error {
 		return errs.NotFound("User", "userID", id)
 	}
 
-	userUUID, err := uuid.Parse(id)
-	if err != nil {
-		print(err.Error())
-		return err
-	}
+	if flag == "id" {
```

Comment:
what is the purpose of this branch? have nesting when it makes sense, if this is the main path, why add an extra indentation?